src/tree/EnvironmentTreeItem.ts

```diff
@@ -133,9 +132,7 @@ export class EnvironmentTreeItem extends AzExtParentTreeItem implements IAzureRe
         await window.withProgress({ location: ProgressLocation.Notification, title: deleting }, async (): Promise<void> => {
             ext.outputChannel.appendLog(deleting);
             const client: WebSiteManagementClient = await createWebSiteClient([context, this]);
-            const deleteResponse = await client.staticSites.deleteStaticSiteBuild(this.parent.resourceGroup, this.parent.name, this.buildId);
-            await pollAzureAsyncOperation(context, deleteResponse, this.subscription);
-
+            await client.staticSites.beginDeleteStaticSiteBuildAndWait(this.parent.resourceGroup, this.parent.name, this.buildId);
```

Comment:
Did you verify that this doesn't need the polling anymore?

If so, I think you can get rid of the util function `pollAzureAsyncOperation`.